locator/token_metadata.cc

```diff
@@ -830,6 +838,12 @@ future<> token_metadata_impl::update_pending_ranges(
                 if (!pending_endpoints.empty()) {
                     add_mapping(migration_info.pending_endpoints, std::move(pending_endpoints));
                 }
+
+                // in order not to waste memory, we update read_endpoints only if the
+                // new endpoints differs from the old one
+                if (_read_new_requested && new_endpoints.get_vector() != old_endpoints.get_vector()) {
```

Comment:
Maybe we should compare sets for safety instead of vectors - since comparing vectors depends on the order of elements.

However since this is only an optimization, it doesn't matter that much.

Reply:
Comparing vectors is more lightweight. They are populated by the same function, the order must be the same.